lib/OpenQA/Task/Job/FinalizeResults.pm

```diff
@@ -0,0 +1,57 @@
+# Copyright (C) 2020 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Task::Job::FinalizeResults;
+use Mojo::Base 'Mojolicious::Plugin';
+
+sub register {
+    my ($self, $app) = @_;
+    $app->minion->add_task(finalize_job_results => sub { _finalize_results($app, @_); });
+    $app->minion->add_task(cleanup_job_results  => sub { _cleanup_results($app, @_); });
+}
+
+sub _finalize_results {
+    my ($app, $task, $job_id) = @_;
+    my $guard = $app->minion->guard("finalize_job_results_for_$job_id", 86400);
+    return $task->finish('Blocked by another finalize task') unless $guard;
+
+    my $job    = $app->schema->resultset('Jobs')->find({id => $job_id});
+    my $errors = 0;
+
+    for my $module (OpenQA::Schema::Result::JobModules::job_modules($job)) {
```

Comment:
The function no longer exists, this should probably just be `$job->modules->all`.